### PR TITLE
[PyOV] Extend Python API with `Squeeze-15`

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - This PR includes commits from https://github.com/openvinotoolkit/openvino/pull/26995

### Tickets:
 - CVS-154024

### CODE REVIEW

File: src/core/include/openvino/op/squeeze.hpp [hunk 1]
```diff
@@ -4,38 +4,75 @@
...
 
     Squeeze();
+    /// \brief Constructs a squeeze v0 operation.
+    ///
+    /// \param data Input tensor with data
+    Squeeze(const Output<Node>& data);
+    /// \brief Constructs a squeeze v0 operation.
+    ///
+    /// \param data Input tensor with data
+    /// \param axis The axis along which to squeeze the input tensor.
```
Comment (rkazants) [inline src/core/include/openvino/op/squeeze.hpp:30]:
```suggestion
    /// \param axes The axes along which to squeeze the input tensor.
```
axis - is just one, axes - can be multiple

File: src/core/include/openvino/op/squeeze.hpp [hunk 2]
```diff
@@ -4,38 +4,75 @@
...
+
+    Squeeze();
+    /// \brief Constructs a squeeze v15 operation.
+    ///
+    /// \param data Input tensor with data
     Squeeze(const Output<Node>& data);
+    /// \brief Constructs a squeeze v15 operation.
+    ///
+    /// \param data Input tensor with data
+    /// \param axis The axis along which to squeeze the input tensor.
```
Comment (rkazants) [inline src/core/include/openvino/op/squeeze.hpp:59]:
```suggestion
    /// \param axes The axes along which to squeeze the input tensor.
```

File: src/bindings/python/tests/test_graph/test_squeeze.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    axes = [2, 4]
+    model = ov_opset1.squeeze(parameter_data, axes)
+
+    assert model.get_type_name() == "Squeeze"
+    assert model.get_output_size() == 1
+    assert list(model.get_output_shape(0)) == [1, 2, 3, 1]
+
+
+@pytest.mark.parametrize(("input_shape", "axes", "allow_axis_skip", "expected_shape"), [
+    ((1, 2, 1, 3, 1, 1), [2, 4], True, [1, 2, 3, 1]),
```
Comment (rkazants) [inline src/bindings/python/tests/test_graph/test_squeeze.py:23]: what will be if axis corresponds to static dimension not equal to `1` for `allow_axis_skip` True and `False`?

Comment (p-wysocki) [reply src/bindings/python/tests/test_graph/test_squeeze.py:23]: In that case the dimension will remain unchanged, I just checked on master. @barnasm1 could you please confirm?

**Review by praasz — APPROVED** [review]

**Pull request merged**